### PR TITLE
Update dogfooding.md for latest CLI changes

### PR DESCRIPTION
**Pull request by eerhardt**
`dotnet new` in the 2.0 CLI handles most of these manual steps now.

@ericstj @weshaggard @terrajobst @danmosemsft

### CODE REVIEW

File: Documentation/project-docs/dogfooding.md [hunk 1]
```diff
@@ -13,59 +13,16 @@ sure to consult this document often.
...
-      affect other projects on your machine and you probably don't want that.
-    - Instead, add a `nuget.config` that is local to your project. You can
-      just put it next to the `.csproj` file.
-      See the [NuGet docs](https://docs.nuget.org/ndocs/consume-packages/configuring-nuget-behavior)
-      for details.
-
-    ```xml
-    <configuration>
-      <packageSources>
-        <add key="CoreFX" value="https://dotnet.myget.org/F/dotnet-core/api/v3/index.json" />
```
Comment (weshaggard) [inline Documentation/project-docs/dogfooding.md]: Does it actually create a nuget.config to our myget feed?

Comment (eerhardt) [inline Documentation/project-docs/dogfooding.md]: yes.

Comment (danmosemsft) [inline Documentation/project-docs/dogfooding.md]: For my education, how is this done? The installer edits this file?

Comment (eerhardt) [inline Documentation/project-docs/dogfooding.md]: The NuGet.config file is simple, there's not need to change its contents:

https://github.com/dotnet/cli/tree/master/src/dotnet/commands/dotnet-new/CSharp_Console

It's the .csproj that is a little harder.  We add a template parameter that gets replaced during `dotnet new`:

https://github.com/dotnet/cli/blob/master/src/dotnet/commands/dotnet-new/Program.cs#L69-L77

**Pull request merged**